tests/serialized_test_data_generation/submit_job.sh

```diff
@@ -1,12 +1,37 @@
 #!/bin/bash
 
+# stop on all errors
+set -e
+
+# setup run directory
 RUNDIR=/rundir
+cp /FV3/fv3.exe $RUNDIR/fv3.exe
+
+# setup directory for serialized data
 TEST_DATA_DIR=$RUNDIR/test_data
+mkdir -p $TEST_DATA_DIR
+
+# setup environment
+
+# run the model (deduce MPI configuration from config file)
+cd $RUNDIR
 NUM_RANKS=`python ${RUNDIR}/count_ranks.py $RUNDIR/fv3config.yml`
 ulimit -s unlimited
-cp /FV3/fv3.exe $RUNDIR/fv3.exe
-mkdir -p $TEST_DATA_DIR
+echo ">>> Running model, inspect stdout.out / stderr.out in rundir for details"
+mpirun -l -np $NUM_RANKS $RUNDIR/fv3.exe 2>$RUNDIR/stderr.out 1>$RUNDIR/stdout.out
```

Comment:
it's great to write the output to a stdout/stderr files, but can we then echo it in the jenkins plan? When it fails we can't tell why because they are being swallowed into these files

Reply:
I changed it so stdout/stderr are echoed only in case of a failure. This keeps Jenkins logs slim but has the information you need. In case the run is successful, the logs are pushed to the GCP bucket and can be inspected there.